metadata/metadata.go

```diff
@@ -82,6 +83,44 @@ func Pairs(kv ...string) MD {
 	return md
 }
 
+// Validate returns error if md is not valid
+// There are check items:
+// - header names contain one or more characters from this set [0-9 a-z _ - .]
+// - if the header-name ends with a "-bin" suffix, the header-value could contain an arbitrary octet sequence. So no real validation required here.
+// - if header-name does not end with a "-bin" suffix, header-value should only contain one or more characters from the set ( %x20-%x7E ) which includes space and printable ASCII.
+func (md MD) Validate() error {
```

Comment:
Move this to an internal package (maybe `internal/metadata`?)
Then this cannot be method. A `func Validate(md MD) error` should be fine, too.